### PR TITLE
Error if '--sif-id' is 0

### PR DESCRIPTION
**Pull request by WestleyK**
## Description of the Pull Request (PR):

If the user passed `--sif-id 0`, sinse that is the default
`verify`/`sign` behavior, it will seem like that flag has no effect
(witch it does not), it should fail as an invalid value.

There are probably better ways to solve this, but this is what I came up
with. Open to suggestions.

### This fixes or addresses the following GitHub issues:

 - Fixes #

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: cmd/internal/cli/sign.go [hunk 1]
```diff
@@ -104,17 +103,29 @@ var SignCmd = &cobra.Command{
...
 func doSignCmd(cpath, url string) error {
-	if sifGroupID != 0 && sifDescID != 0 {
-		return fmt.Errorf("only one of -i or -g may be set")
+	if sifGroupID != -1 && sifDescID != -1 {
+		sylog.Fatalf("only one of -i or -g may be set")
+	}
+
+	// If the user used '0' as descriptor id, then fail as
+	// 0 is invalid.
+	if sifGroupID == -1 && sifDescID == 0 {
```
Comment (cclerget) [inline cmd/internal/cli/sign.go]: Looks like both `sifGroupID` and `sifDescID` can't be 0

File: cmd/internal/cli/sign.go [hunk 2]
```diff
@@ -87,7 +87,6 @@ var SignCmd = &cobra.Command{
 	Args:                  cobra.ExactArgs(1),
 
 	Run: func(cmd *cobra.Command, args []string) {
-
 		// args[0] contains image path
 		fmt.Printf("Signing image: %s\n", args[0])
 		if err := doSignCmd(args[0], keyServerURI); err != nil {
```
Comment (cclerget) [inline cmd/internal/cli/sign.go]: while we are, can we remove `keyServerURI` which is global and maybe replace `sylog.Errorf/os.Exit` by `sylog.Fatalf()` unless there is need to return a specific exit code ?

File: cmd/internal/cli/sign.go [hunk 3]
```diff
@@ -104,17 +103,29 @@ var SignCmd = &cobra.Command{
 }
 
 func doSignCmd(cpath, url string) error {
-	if sifGroupID != 0 && sifDescID != 0 {
-		return fmt.Errorf("only one of -i or -g may be set")
+	if sifGroupID != -1 && sifDescID != -1 {
+		sylog.Fatalf("only one of -i or -g may be set")
+	}
+
```
Comment (cclerget) [inline cmd/internal/cli/sign.go:105]: you could use cobra by passing `cmd *cobra.Command` to `doSignCmd` and have something like

```
if cmd.Flag(signSifGroupIDFlag.Name).Changed() && sifGroupID == 0 {
    sylog.Fatalf("invalid group id")
}
if cmd.Flag(signSifDescSifIDFlag.Name).Changed() && sifDescID == 0 {
    sylog.Fatalf("invalid descriptor id")
}
if cmd.Flag(signSifDescIDFlag.Name).Changed() && sifDescID == 0 {
    sylog.Fatalf("invalid descriptor id")
}
```

**Review by cclerget — CHANGES_REQUESTED** [review]
Left suggestions about using cobra instead, request changes mostly for groupid must be > 0

File: cmd/internal/cli/sign.go (hunk 2 above)
Comment (WestleyK) [reply cmd/internal/cli/sign.go]: I just removed `keyServerURI` altogether, does not seem like it was used anywhere here.

Comment (WestleyK) [reply cmd/internal/cli/sign.go]: Also removed the error return, to keep it consistent with `verify.go`.

**Review by cclerget — APPROVED** [review]
LGTM. Thanks !

**Pull request merged**